semgrep/semgrep/rule_match.py

```diff
@@ -76,7 +76,7 @@ def end(self) -> Dict[str, Any]:
         return self._end
 
     @property
-    def lines(self) -> Iterable[str]:
+    def lines(self) -> List[str]:
```

Comment:
This was necessary to perform list indexing in `rule_match_nosem`. The `typing` docs also suggest using `List` for return types and `Iterable` for arguments: https://docs.python.org/3/library/typing.html#typing.List